src/model/user.rs

```diff
@@ -446,6 +448,43 @@ pub struct User {
     /// change if the username+discriminator pair becomes non-unique.
     #[serde(rename = "username")]
     pub name: String,
+    /// the public flags on a user's account
+    pub public_flags: UserPublicFlags,
+}
+
+bitflags! {
+    /// User public flags
+    #[derive(Default, Deserialize, Serialize)]
+    pub struct UserPublicFlags: u32 {
+        /// User's flag as none
+        const NONE = 0b00000000_00000000_00000000_00000000;
+        /// User's flag as discord employee
+        const DISCORD_EMPLOYEE = 0b00000000_00000000_00000000_00000001;
+        /// User's flag as partnered server owner
+        const PARTNERED_SERVER_OWNER = 0b00000000_00000000_00000000_00000010;
+        /// User's flag as hypesquad events
+        const HYPESQUAD_EVENTS = 0b00000000_00000000_00000000_00000100;
+        /// User's flag as bug hunter level 1
+        const BUG_HUNTER_LEVEL_1 = 0b00000000_00000000_00000000_00001000;
+        /// User's flag as house bravery
+        const HOUSE_BRAVERY = 0b00000000_00000000_00000000_01000000;
+        /// User's flag as house brilliance
+        const HOUSE_BRILLIANCE = 0b00000000_00000000_00000000_10000000;
+        /// User's flag as house balance
+        const HOUSE_BALANCE = 0b00000000_00000000_00000001_00000000;
+        /// User's flag as early supporter
+        const EARLY_SUPPORTER = 0b00000000_00000000_00000010_00000000;
+        /// User's flag as team user
+        const TEAM_USER = 0b00000000_00000000_00000100_00000000;
+        /// User's flag as system
+        const SYSTEM = 0b00000000_00000000_00010000_00000000;
+        /// User's flag as bug hunter level 2
+        const BUG_HUNTER_LEVEL_2 = 0b00000000_00000000_01000000_00000000;
+        /// User's flag as verified bot
+        const VERIFIED_BOT = 0b00000000_00000001_00000000_00000000;
+        /// User's flag as early verified bot developer
+        const EARLY_VERIFIED_BOT_DEVELOPER = 0b00000000_00000010_00000000_00000000;
+    }
 }
```

Comment:
This should be implemented like it is for [`Permissions`](https://github.com/serenity-rs/serenity/blob/current/src/model/permissions.rs#L207-L319) and [other bitflags structs](https://github.com/serenity-rs/serenity/blob/1ab52a056f94760a7436927c98c08ea27da3a0ad/src/model/channel/message.rs#L1121-L1138).

Reply:
teach me how to use bitflags in the right way

Reply:
I apologise that I haven't gotten to you in a while, but from the latest commit, you got it right. I do have a small nitpick, but otherwise good job!